### PR TITLE
Document what to do if a merged PR has a bug that was missed during PR reviews

### LINKED ISSUE #449: Help Wanted: Document what to do if a merged PR has a bug that was missed during PR reviews
Mattermost user `jason.blais` from https://community-release.mattermost.com has requested the following be documented:

```
If we were to find something wrong with a merged PR (made by us), what is the standard procedure? Create a new issue and new PR to solve the problem? Somehow reopen the PR?

Answer: I'd recommend creating a GitHub issue + PR that fixes the issue. If it's a major problem, then reverting the changes instead (and reopening a PR with the fix) is the safer approach.
```

See the original post [here](https://community-release.mattermost.com/_redirect/pl/3mo4k3kk6bypdje4ynqzdji19h).

_This issue was generated from [Mattermost](https://mattermost.com) using the [Doc Up](https://github.com/jwilander/mattermost-plugin-docup) plugin._

### PR DESCRIPTION
**Pull request by TomerPacific**
#### Summary
We want to present a process for handling a scenario where a PR was merged and the community found out that there was a mistake with it.

Following the discussion in the issue, I created a new page called, **Merged PR Scenario** under the good decisions section inside of the contribute section.

This page describes what should be done depending on the type of the feature merged in the PR (either low or high impact).

#### Ticket Link
Fixes #449

### CODE REVIEW

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+title: "Merged PR Scenario"
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:2]: Let's reword this new page title to describe the reader's circumstance. What scenario would bring a user to this page to learn more?

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+heading: "Merged PR Scenario"
+weight: 2
+---
+
+This document's aim is to explain what process should be in place in the following scenario:
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:7]: Let's introduce the scenario less formally as something along the lines of "This page describes the process to follow when someone notices a mistake in a merged pull request (PR)."

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+1. A contributor (either staff or community member) submits a PR, it is reviewed and merged into the codebase
+2. Sometime later, the community notices a mistake with the PR
+
+Question is, what should we, as a community, do? That depends on the scope of the changes in the PR that was merged.
+
+## Low Impact
+A low impact PR might mean that it affected:
+- Some non-critical functionality
+- It does not affect users in a substantial way
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:17]:
```suggestion
- It doesn't affect users in a substantial way 
```

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+## Low Impact
+A low impact PR might mean that it affected:
+- Some non-critical functionality
+- It does not affect users in a substantial way 
+
+If this is the case, do the following:
+
+1. Create an issue for this
+2. Mark it according to it's priority
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:22]:
```suggestion
2. Mark it according to its priority.
```

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+Question is, what should we, as a community, do? That depends on the scope of the changes in the PR that was merged.
+
+## Low Impact
+A low impact PR might mean that it affected:
+- Some non-critical functionality
+- It does not affect users in a substantial way 
+
+If this is the case, do the following:
+
+1. Create an issue for this
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:21]:
```suggestion
1. Capture details in an issue.
```

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+2. Mark it according to it's priority
+3. Would be best to assign it to the person who introduced the issue in the first place
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:23]:
```suggestion
3. Ideally, assign the issue to the person who created the PR.
```

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+---
+
+This document's aim is to explain what process should be in place in the following scenario:
+
+1. A contributor (either staff or community member) submits a PR, it is reviewed and merged into the codebase
+2. Sometime later, the community notices a mistake with the PR
+
+Question is, what should we, as a community, do? That depends on the scope of the changes in the PR that was merged.
+
+## Low Impact
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:14]:
```suggestion
## Low impact issues
```

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+- Some non-critical functionality
+- It does not affect users in a substantial way 
+
+If this is the case, do the following:
+
+1. Create an issue for this
+2. Mark it according to it's priority
+3. Would be best to assign it to the person who introduced the issue in the first place
+
+## High Impact
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:25]:
```suggestion
## High impact issues
```

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+A high impact PR represents something that caused an incident or might well be causing one and we caught it early.
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:26]:
```suggestion
A high impact PR represents something that has or will result in a customer incident.
```

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+If this is the case, there are two scenarios:
+
+1. The feature introduced in the PR is handled by a feature flag
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:30]: Please end all sentences with a period.

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+## High Impact
+A high impact PR represents something that caused an incident or might well be causing one and we caught it early.
+
+If this is the case, there are two scenarios:
+
+1. The feature introduced in the PR is handled by a feature flag 
+2. The feature introduced in the PR is **not** handled by a feature flag 
+
+For scenario 1, if it is not affecting other functionality, turn that feature flag off so the feature will be disabled.
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:33]:
```suggestion
For scenario 1, if it's not affecting other functionality, turn that feature flag off to disable the feature.
```

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+For scenario 2:
+
+1. Revert the changes introduced in the original PR
+2. Let the original person who worked on the PR know this so they can work on a proper fix for their PR
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:38]:
```suggestion
2. Notify the person who worked on the PR know so they can work on a proper fix for their PR
```

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+3. Reintroduce the change and have it go through the regular PR cycle
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:39]:
```suggestion
3. Reintroduce the change through the regular PR cycle
```

**Review by cwarnermm — COMMENTED** [review]
Thanks, @TomerPacific! This is a great documentation update! I've provided feedback to more closely align this content with our writing guidelines and style guide.

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md (hunk 1 above)
Comment (TomerPacific) [reply site/content/contribute/good-decisions/fixing-pr-mistakes.md]: Reworded title.

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md (hunk 2 above)
Comment (TomerPacific) [reply site/content/contribute/good-decisions/fixing-pr-mistakes.md]: Done.

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md (hunk 10 above)
Comment (TomerPacific) [reply site/content/contribute/good-decisions/fixing-pr-mistakes.md]: Done.

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
+---
+title: "When A Merged PR Results In A Bug"
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:2]:
```suggestion
title: "When A merged PR results in a bug"
```

**Review by cwarnermm — APPROVED** [review]
Good updates and clarifications for the page visitor, @TomerPacific! I've applied a few sentence case updates.

File: site/content/contribute/good-decisions/fixing-pr-mistakes.md [hunk 15]
```diff
@@ -0,0 +1,39 @@
+---
+title: "When a merged PR results in a bug"
+heading: "When A Merged PR Results In A Bug"
```
Comment (cwarnermm) [inline site/content/contribute/good-decisions/fixing-pr-mistakes.md:3]:
```suggestion
heading: "When a merged PR results in a bug"
```

**Review by neflyte — APPROVED** [review]

**Pull request merged**